File: src/Entity/Category.php
<?php

namespace App\Entity;

use App\Repository\CategoryRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: CategoryRepository::class)]
class Category
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name:'categoryId')]
    private ?int $categoryId = null;

    #[ORM\Column(length: 255)]
    /**
 * @Assert\NotBlank(message="Don't leave it blank")
 */
    private ?string $Product_Type = null;

    #[ORM\Column(length: 255)]
    /**
 * @Assert\NotBlank(message="Don't leave it blank")
 */
    private ?string $Product_Source = null;

    #[ORM\OneToMany(mappedBy: 'Category_Key', targetEntity: Product::class)]
    private Collection $products;

    public function __construct()
    {
        $this->products = new ArrayCollection();
    }

    public function getcategoryId(): ?int
    {
        return $this->categoryId;
    }

    public function getProductType(): ?string
    {
        return $this->Product_Type;
    }

    public function setProductType(string $Product_Type): static
    {
        $this->Product_Type = $Product_Type;

        return $this;
    }
    public function __toString()
    {
        return $this->Product_Type;
    }
    public function getProductSource(): ?string
    {
        return $this->Product_Source;
    }

    public function setProductSource(string $Product_Source): static
    {
        $this->Product_Source = $Product_Source;

        return $this;
    }

    /**
     * @return Collection<int, Product>
     */
    public function getProducts(): Collection
    {
        return $this->products;
    }

    public function addProduct(Product $product): static
    {
        if (!$this->products->contains($product)) {
            $this->products->add($product);
            $product->setCategoryKey($this);
        }

        return $this;
    }

    public function removeProduct(Product $product): static
    {
        if ($this->products->removeElement($product)) {
            // set the owning side to null (unless already changed)
            if ($product->getCategoryKey() === $this) {
                $product->setCategoryKey(null);
            }
        }

        return $this;
    }
}

File: src/main/java/entities/Category.java
package entities;

public class Category {
  int CategoryId;
  String Product_Type;
  String Product_Source;

    public Category() {
    }
    public Category(int categoryId, String product_Type, String product_Source) {
        this.CategoryId = categoryId;
        this.Product_Type = product_Type;
        this.Product_Source = product_Source;
    }
    public Category(String product_Type, String product_Source) {
        this.Product_Type = product_Type;
        this.Product_Source = product_Source;
    }


       public int getCategoryId() {
           return CategoryId;
       }

       public void setCategoryId(int categoryId) {
           CategoryId = categoryId;
       }

       public String getProduct_Type() {
           return Product_Type;
       }

       public void setProduct_Type(String product_Type) {
           Product_Type = product_Type;
       }

       public String getProduct_Source() {
           return Product_Source;
       }

       public void setProduct_Source(String product_Source) {
           Product_Source = product_Source;
       }


    @Override
    public String toString() {
        return "Category{" +
                "CategoryId=" + CategoryId +
                ", Product_Type='" + Product_Type + '\'' +
                ", Product_Source='" + Product_Source + '\'' +
                '}';
    }
}
